src/tests/test_sexual_behaviour.py

```diff
@@ -104,3 +106,225 @@ def test_initial_sex_behaviour_groups():
+    assert np.isclose(SBM.rred_population, SBM.yearly_risk_change["1990s"], 0.001)
+    pop.date = date(1998, 1, 1)
+    SBM.update_rred(pop)
+    assert np.isclose(SBM.rred_population, SBM.yearly_risk_change["1990s"]**3, 0.001)
+    pop.date = date(2000, 1, 1)
+    SBM.update_rred(pop)
+    assert np.isclose(SBM.rred_population, SBM.yearly_risk_change["1990s"]**5, 0.001)
+    pop.date = date(2010, 1, 1)
+    SBM.update_rred(pop)
+    assert np.isclose(SBM.rred_population, SBM.yearly_risk_change["1990s"]**5, 0.001)
+    pop.date = date(2011, 1, 1)
+    SBM.update_rred(pop)
+    assert np.isclose(SBM.rred_population,
+                      SBM.yearly_risk_change["1990s"]**5 *
+                      SBM.yearly_risk_change["2010s"], 0.001)
+    pop.date = date(2020, 1, 1)
+    SBM.update_rred(pop)
+    assert np.isclose(SBM.rred_population,
+                      SBM.yearly_risk_change["1990s"]**5 *
+                      SBM.yearly_risk_change["2010s"]**10, 0.001)
+    pop.date = date(2022, 1, 1)
+    SBM.update_rred(pop)
+    assert np.isclose(SBM.rred_population,
+                      SBM.yearly_risk_change["1990s"]**5 *
+                      SBM.yearly_risk_change["2010s"]**11, 0.001)
+
+
+def test_rred_personal():
+    N = 100
+    pop = Population(size=N, start_date=date(1989, 1, 1))
+    # Count how many times we initialise with each threshold 0.3, 0.5, 0.7
+    count03, count05, count07 = (0, 0, 0)
+    for i in range(100):
+        SBM = SexualBehaviourModule()
+        SBM.init_risk_factors(pop.data)
+        risk_count = sum(selector(pop.data, rred_personal=(operator.lt, 1)))
+        count03 += (0.2 < (risk_count/N) < 0.4)  # check consistency with threshold
+        count05 += (0.4 < (risk_count/N) < 0.6)  # check consistency with threshold
+        count07 += (0.6 < (risk_count/N) < 0.8)  # check consistency with threshold
+        checks = np.array([x == 1 or x == 1e-5 for x in pop.data["rred_personal"]])
+        assert np.all(checks)  # Check that we only get valid rred values
+    assert(1/6 < count03/100 < 1/2)  # check frequency of threshold from initialisations
+    assert(1/6 < count05/100 < 1/2)  # check frequency of threshold from initialisations
+    assert(1/6 < count07/100 < 1/2)  # check frequency of threshold from initialisations
+
+
+def test_rred_age():
+    N = 11
+    pop = Population(size=2*N, start_date=date(1989, 1, 1))
+    ages = np.array([12, 17, 22, 27, 32, 37, 42, 47, 52, 57, 62]*2)
+    pop.data["age"] = ages
+    pop.data["sex"] = np.array([SexType.Male]*N + [SexType.Female]*N)
+    pop.sexual_behaviour.init_risk_factors(pop.data)
+    # select a particular risk matrix
+    risk_factors = np.array(pop.data["rred_age"])
+    print(risk_factors)
+    assert risk_factors[0] == 1  # no update for under 15
+    assert risk_factors[11] == 1
+    expected_risk_male = pop.sexual_behaviour.age_based_risk.transpose()[0]
+    expected_risk_female = pop.sexual_behaviour.age_based_risk.transpose()[1]
```

Comment:
A bit shorter (I think this will be a 2D array, right?):
```suggestion
    expected_risk_male = pop.sexual_behaviour.age_based_risk[:, 0]
    expected_risk_female = pop.sexual_behaviour.age_based_risk[:, 1]
```
or (arrays are unpacked by row):
```suggestion
    expected_risk_male, expected_risk_female = pop.sexual_behaviour.age_based_risk.T
```